aggregator/src/app/BundleRouter.ts

```diff
@@ -24,14 +24,29 @@ export default function BundleRouter(bundleService: BundleService) {
   router.get(
     "bundleReceipt/:hash",
     async (ctx) => {
+      const pendingBundle = await bundleService.lookupBundle(ctx.params.hash!);
+      if (pendingBundle) {
+        ctx.response.status = 202;
```

Comment:
I don't think 202 is appropriate here. It's the *bundle* that's been accepted, not the receipt. Also 'accepted' is supposed to refer to the current request, not the resource you're asking about. [More info](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202).

(Possibly 202 is appropriate for the original POST to /bundle, but I'm not actually a fan of worrying about these HTTP nuances.)

On a practical note, if we keep 404 here we can still also provide the `submitError`, and this way it wouldn't be a breaking change. Wdyt?

Reply:
Yeah, I get what you're saying.  I do like the idea of not doing a breaking change, as there are a good amount of things that have to be updated at this point.

However, I think the `/bundleReceipt` API call is often used as a status check for a bundle.  So my question would be, what should we expect to be returned from this API call?  With the approach you recommended, we would keep a 200 status as is (i.e. returns the bundle receipt) and pass the submit error in a 404 response body.  That would be fine by me - I think the only thing we aren't communicating explicitly is the bundle status. For example, we could communicate a pending or a failed state (ex: even if it failed once, we may still retry it).  Maybe we don't care about that though or it should be a different API endpoint.

@voltrevo What do you think?  I'd be happy to hop on a call tomorrow when I get online if you are still around.  Also, @jacque006  not sure if you have any thoughts?

Reply:
I'm not sure we're on the same page here. Definitely sounds like a call is in order.

> With the approach you recommended, we would keep a 200 status as is (i.e. returns the bundle receipt) and pass the submit error in a 404 response body.

Did you mean to say 'keep a 200 status'? It seems wrong, and also contradicts the second half of this sentence. At least my reading of it.

To rephrase my suggestion, I think if the bundle is pending, then `/bundleReceipt` should 404, since there isn't a bundle receipt, but we can also put the submitError in the body of this response. Meaning: "No, there's no receipt, and here's why".

Reply:
Ahh my response was poorly worded.  We are on the same page.  When I said keep the 200 status, I really meant get rid of my logic to add the 202.  You worded it better than I did - that is what I meant to say.

Reply:
Yep, we can discuss offline. In general, I have seen services go one of two ways:

- Use a variety/specific HTTP codes to communicate the state of a resource (200/201/202/403/404/418)
- Use a limited number of status codes (200/403/404) and use a `status` field or similar to communicate the state of the resource (pending/failed/succeeded). Some Google Cloud APIs go this route

I think either is fine as long as we are consistent with that pattern on our endpoints and document the possible responses in code. Hopefully most consumers use our clients which can abstract a lot of that away.